TileSystem/Implementation/TwoDimension/Area.cs

```diff
@@ -26,12 +30,15 @@ public class Area : IArea
 		public event EventHandler<TileRemovedArgs> TileRemoved;
 
 		// Representation in the system
-		public string Type { get; private set; }
-		public string Variation { get; private set; }
+		public string Type { get; protected set; }
+		public string Variation { get; protected set; }
 
 		// Location in the system
 		public ILevel Level { get; private set; }
-		public IPosition2D Position { get; private set; }
+		public IPosition Position
```

Comment:
The only problem is where it's used you will have to cast, but to keep the internals clean it is actually better in my opinion, but when you put samples together we will really know.